### DISCUSSION TITLE
On a virtual machine, Mac OS Sonoma does not show the running application

### DISCUSSION THREAD

**Discussion by konfringo17** [flet-dev/flet#3342] [General]
**Hi all!**
1. The latest version of the Mac OS Sonoma virtual machine is installed
2. Installed paycharm and flet. 
3. Wrote simple code
4. I launch the application - the flet icon appears, but nothing is displayed on the screen...
5. After launching the application, I click on the icon at the bottom, and nothing happens.
On another virtual machine Mac OS Catalina, the window opened, but it was empty.
![flet](https://github.com/flet-dev/flet/assets/170368845/9cd0187d-50f3-4c17-a5b1-1b8c0f86bd43)

Maybe my flet is installed in the wrong place?
This is the first time in my life I'm using Mac OS, I made a cool application on Android using flet, but there were problems on Mac OS.
I don’t understand what to do, and it’s very difficult to buy a MacBook for now.
Maybe someone has encountered this?
I launched Tkinter, it opens and works relatively fine.

**Comment by FeodorFitsner** [comment]
What virtualization software do you use? It could be your case: https://stackoverflow.com/a/75354916

Flet is based on Flutter which since recently requires Metal support. I can run Flet apps in the latest Parallels VMs without any issues.

**Reply by konfringo17** [reply]
For virtualization I use VMware Workstation Pro - I installed it on Windows
I only have Windows, I thought that I would install a virtual machine, install a Mac OS on it and be able to compile the .ipa file of my application. It’s not working yet)

Can I install Parallels on Windows and install Mac OS on it?

I read from the link that OpenGl was removed from virtual machines and that because of this, flet doesn’t seem to be displayed.

In fact, I was and agreed that I would just copy my code into the Mac OS and just compile it, without previewing the application, at least that way to begin with

**Reply by FeodorFitsner** [reply]
Um, well, my guess is that (according to Apple licensing and maybe implementation) macOS VMs could be run on macOS hosts only. I don't know what Parallels/VMWare are offering on Windows though.

**Reply by FeodorFitsner** [reply]
You can rent a cheap Mac here: https://www.macminivault.com/dedicated-mac-mini/ - great company, great service, great pricing (we use them).
